cdap-api/src/main/java/io/cdap/cdap/api/metrics/Metrics.java

```diff
@@ -34,5 +36,18 @@
    * @param value The value to be set.
    */
   void gauge(String metricName, long value);
+
+  /**
+   * Get the child {@link Metrics} by adding specified tags to the current metrics context.
+   * @param tags to be added to the current metrics context
+   * @return an instance of {@link Metrics} which can be used to collect metrics with new child context.
+   * @throws IllegalArgumentException if any of the supplied tag name already exists in the current metrics context
+   */
+  Metrics child(Map<String, String> tags) throws IllegalArgumentException;
```

Comment:
generally we don't declare throws [? extends RuntimeException], we just put it in the javadoc.

Reply:
ah right.. removed.